straxen/url_config.py

```diff
@@ -47,8 +47,13 @@ def __init__(self, cache=0, **kwargs):
             self.final_type = self.type
             self.type = OMITTED # do not enforce type on the URL
         if cache:
-            maxsize = cache if isinstance(cache, int) else None
-            self.dispatch = lru_cache(maxsize)(self.dispatch)
+            cache_len = cache if isinstance(cache, int) else 1_000_000
+            cache = straxen.CacheDict(cache_len=cache_len)
+            _CACHES[id(self)] = cache
```

Comment:
what is the unit of this? Because 1e6 items/config seems huge

Reply:
yeah its number of items. its supposed to be effectively unbounded when you set it to True. But if people tend to just set the cache to True instead of specifying a specific size, then maybe we can set a smaller default value.

Reply:
òkay, I actually thought `True` would mean `int(True)->1` 😂 .

Just to be sure, each run gets it's own instance right and thereby it's own configs (id)? So while the number here basically reflects how many runs (assuming you don't change your config) you can cache for this given option?

Reply:
ah you are right! I forgot True is an instance of int, so yes it will give you a cache with size 1. fixed that.

> Just to be sure, each run gets it's own instance right and thereby it's own configs (id)? So while the number here basically reflects how many runs (assuming you don't change your config) you can cache for this given option?

each plugin class gets its own instance of URLConfig, each instance of URLConfig instance gets its own cache and each unique URL gets its own cache entry. 
So if the URL depends on the run_id then yes, its how many runs are cached, if the URL doesn't depend on the run_id then you can cache much more. If the URL depends on some plugin attribute that is set to a different value for each chunk then that's the maximum number of chunk values kept in memory at any given time etc. your URL can depend on arbitrary plugin attributes that can change during computation. But this feature probably wont be used in practice, most likely it will just depend on run_id.

Its important to mention that its an LRU cache, so its not that if the cache fills up you get an error or something. The least recently used value is just evicted and replaced by the most recently accessed value.

Reply:
> If the URL depends on some plugin attribute that is set to a different value for each chunk then that's the maximum number of chunk values kept in memory at any given time etc. your URL can depend on arbitrary plugin attributes that can change during computation. But this feature probably wont be used in practice, most likely it will just depend on run_id.

Right, then we should really make sure people don't set a cache at all, there is almost never any reason to assume you need a value more than once for a same chunk within the same plugin 😉